akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -129,6 +129,12 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
 
       override protected def logSource: Class[_] = if (isServer) classOf[Http2ServerDemux] else classOf[Http2ClientDemux]
 
+      def frameOutFinished(): Unit = {
+        // make sure we clean up/fail substreams with a custom failure before stage is canceled
+        // and substream autoclean kicks in
+        shutdownStreamHandling()
```

Comment:
`Http2Multiplexer` does not know that `this` is also a `Http2StreamHandling` so that's why we need a call back out here to call `shutdownStreamHandling()`